Makefile

```
@@ -111,8 +111,17 @@ staticcheck:
 test:
 	$(GOTEST) -v ./...
 
+.PHONY: test-coverprofile
+test-coverprofile:
+	go test -covermode=count -coverprofile=coverage.out `go list ./... | grep -v integration` -json > test-report.out
```

The Makefile defines `GOTEST` and `GOLIST` as a variables so you could use that here to take advantage of the global settings (if we make any).
Might be worth getting rid of the legacy backticks in the command (https://github.com/koalaman/shellcheck/wiki/Sc2006)

```suggestion
	$(GOTEST) -covermode=count -coverprofile=coverage.out $$($(GOLIST) ./... | grep -v integration) -json > test-report.out
```

---

done